### PR TITLE
[CARBONDATA-489] Fix spark2 decimal issue

### PR DESCRIPTION
**Pull request by scwf**
also added a test suite for decimal

### CODE REVIEW

File: integration/spark2/src/main/scala/org/apache/spark/sql/TableCreator.scala [hunk 1]
```diff
@@ -367,8 +367,7 @@ object TableCreator {
         field.storeType
       )
       case "decimal" => Field(field.column, Some("Decimal"), field.name, Some(null), field.parent,
-        field.storeType, field.precision, field.scale
-      )
+        field.storeType, field.schemaOrdinal, field.precision, field.scale)
```
Comment (scwf) [inline integration/spark2/src/main/scala/org/apache/spark/sql/TableCreator.scala]: this is the bug, here is no field.schemaOrdinal in old version

File: pom.xml [hunk 2]
```diff
@@ -309,9 +309,6 @@
     </profile>
     <profile>
       <id>spark-1.5</id>
-      <activation>
```
Comment (jackylk) [inline pom.xml]: do not modify this

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala [hunk 3]
```diff
@@ -30,14 +30,15 @@ class CarbonDataSourceSuite extends FunSuite with BeforeAndAfterAll {
       .appName("CarbonExample")
       .enableHiveSupport()
       .config(CarbonCommonConstants.STORE_LOCATION,
-        s"examples/spark2/target/store")
+        s"/user/hive/warehouse/store")
```
Comment (jackylk) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala]: Is it ok to hard code this? how about in non-linux?

File: integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonMetastore.scala [hunk 4]
```diff
@@ -125,6 +125,16 @@ class CarbonMetastore(conf: RuntimeConfig, val storePath: String) extends Loggin
...
   }
 
+  def cleanStore(): Unit = {
+    try {
+      val fileType = FileFactory.getFileType(storePath)
+      FileFactory.deleteFile(storePath, fileType)
+    } catch {
+      case e => logError("clean store failed", e)
+    }
+
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonMetastore.scala]: remove empty line

File: core/src/main/java/org/apache/carbondata/core/datastorage/store/impl/FileFactory.java [hunk 5]
```diff
@@ -397,6 +397,38 @@ public static boolean createNewFile(String filePath, FileType fileType) throws I
...
+
+      case LOCAL:
+      default:
+        File file = new File(filePath);
+        return deleteAllFilesOfDir(file);
+    }
+  }
+
+  public static boolean deleteAllFilesOfDir(File path) {
+    if (!path.exists())
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastorage/store/impl/FileFactory.java]: add `{`

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**